service/history/replication/ack_manager.go

```diff
@@ -145,7 +145,7 @@ func (p *ackMgrImpl) GetMaxTaskID() int64 {
 		// use ImmediateTaskMaxReadLevel which is the max task id of any immediate task queues.
 		// ImmediateTaskMaxReadLevel will be the lower bound of new range_id if shard reload. Remote cluster will quickly (in
 		// a few seconds) ack to the latest ImmediateTaskMaxReadLevel if there is no replication tasks at all.
-		return p.shard.GetQueueMaxReadLevel(tasks.CategoryReplication, p.currentClusterName).TaskID
+		return p.shard.GetQueueExclusiveMaxReadLevel(tasks.CategoryReplication, p.currentClusterName).Prev().TaskID
```

Comment:
GetQueueExclusiveMaxReadLevel -> GetQueueExclusiveWatermark? GetQueueExclusiveTaskID?

Reply:
I still think the existing max read level is more accurate. `ExclusiveWatermark` is not very clear about that watermark it is. Same for `TaskID`. 

`ExclusiveHighReadWatermark` probably?